spec/virtual_machine_service_spec.rb

```diff
@@ -111,6 +111,29 @@
     end
   end
 
+  context "operations" do
+    let(:response) { IO.read('spec/fixtures/operations_response.json') }
+
+    before do
+      allow(vms).to receive(:rest_post).and_return(response)
+      allow(response).to receive(:code).and_return(202)
+      allow(response).to receive(:body).and_return('')
+      allow(response).to receive(:headers).and_return(response)
```

Comment:
To avoid misleading, better to append `headers` such as 
```
  let(:response_headers) { IO.read('spec/fixtures/operations_response_headers.json') }
  ...
    allow(response).to receive(:headers).and_return(response_headers)
```
Also please assert `response_code` in headers.